dev/sbt-plugin/src/sbt-test/sbt-plugin/akka-persistence-typed-migration-scala/shopping-cart-lagom-persistence/src/main/scala/com/example/shoppingcart/impl/ShoppingCartEntity.scala

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.example.shoppingcart.impl
+
+import java.time.{Instant, OffsetDateTime}
+
+import akka.Done
+import com.lightbend.lagom.scaladsl.persistence.{AggregateEvent, AggregateEventTag, PersistentEntity}
+import com.lightbend.lagom.scaladsl.persistence.PersistentEntity.ReplyType
+import com.lightbend.lagom.scaladsl.playjson.{JsonSerializer, JsonSerializerRegistry}
+import play.api.libs.json._
+
+import scala.collection.immutable.Seq
+
+/**
+  * This is an event sourced entity. It has a state,
+  * [[com.example.shoppingcart.impl.ShoppingCartState]], which stores the
+  * current shopping cart items and whether it's checked out.
+  *
+  * Event sourced entities are interacted with by sending them commands. This
+  * entity supports three commands, an
+  * [[com.example.shoppingcart.impl.UpdateItem]] command, which is used to
+  * update the quantity of an item in the cart, a
+  * [[com.example.shoppingcart.impl.Checkout]] command which is used to set
+  * checkout the shopping cart, and a [[com.example.shoppingcart.impl.Get]]
+  * command, which is a read only command which returns the current shopping
+  * cart state.
+  *
+  * Commands get translated to events, and it's the events that get persisted
+  * by the entity. Each event will have an event handler registered for it, and
+  * an event handler simply applies an event to the current state. This will be
+  * done when the event is first created, and it will also be done when the
+  * entity is loaded from the database - each event will be replayed to
+  * recreate the state of the entity.
+  *
+  * This entity defines two events, the
+  * [[com.example.shoppingcart.impl.ItemUpdated]] event, which is emitted when
+  * a [[com.example.shoppingcart.impl.UpdateItem]] command is received, and a
+  * [[com.example.shoppingcart.impl.CheckedOut]] event, which is emitted when a
+  * [[com.example.shoppingcart.impl.Checkout]] command is received.
+  */
+class ShoppingCartEntity extends PersistentEntity {
+
+  import play.api.libs.functional.syntax._
+  def naStringSerializer: Format[Option[String]] =
+    implicitly[Format[String]].inmap(
+      str => Some(str).filterNot(_ == "N/A"),
+      maybeStr => maybeStr.getOrElse("N/A")
+    )
+
+  override type Command = ShoppingCartCommand[_]
+  override type Event = ShoppingCartEvent
+  override type State = ShoppingCartState
+
+  /**
+    * The initial state. This is used if there is no snapshotted state to be found.
+    */
```

Comment:
same thing for all code comments: make them relevant in the context of the `scripted` test or remove them.